src/Adapter/Address/CommandHandler/EditCustomerAddressHandler.php

```diff
@@ -48,21 +51,41 @@ final class EditCustomerAddressHandler extends AbstractAddressHandler implements
      * @throws AddressConstraintException
      * @throws CannotUpdateAddressException
      */
-    public function handle(EditCustomerAddressCommand $command): void
+    public function handle(EditCustomerAddressCommand $command): AddressId
     {
         try {
-            $address = $this->getAddressFromCommand($command);
+            $editedAddress = $this->getAddressFromCommand($command);
 
-            if (false === $address->validateFields(false)) {
+            if (false === $editedAddress->validateFields(false)) {
                 throw new AddressConstraintException('Address contains invalid field values');
             }
 
-            if (false === $address->update()) {
-                throw new CannotUpdateAddressException(sprintf('Failed to update address "%s"', $address->id));
+            // The address is used by an order so it is not edited directly, instead a copy is created and
+            if ($editedAddress->isUsed()) {
```

Comment:
@PierreRambaud do you think it's clearer with this naming? Also I thought maybe `$backupAddress` instead of `$copyAddress` would be better?

Reply:
It's not really a backup but a real copy.
But it's clearer now :)